### PR TITLE
Wrapper the vfinance balance query api and provide the balance query …

### PR DESCRIPTION
**Pull request by songrgg**
Wrapper the vfinance balance query api and provide the balance query and order status query api.

1. Wrapper the vfinance balance query api.
2. Provide 'http://api.money.wallstreetcn.com/v1/account/orderhist' for order history query.
3. Provide 'http://api.money.wallstreetcn.com/v1/account/balance' for user balance query.
4. Refactor the service configuration.

### CODE REVIEW

File: apps/FMarketApiVer1/src/FMarketApiVer1/Controllers/AccountController.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    public function initialize() {
+        $this->user = $this->FMAuth->getUser();
+    }
+
+    public function balanceAction()
+    {
+        $userId = $this->user->getId();
+        $balance = VFWallet::queryBalance($userId);
+        if ($balance['result'] === false) {
+            throw new Exception\RuntimeException($balance['errMsg']);
```
Comment (chaiyangyang) [inline apps/FMarketApiVer1/src/FMarketApiVer1/Controllers/AccountController.php]: use const string

File: modules/EvaPay/src/EvaPay/Models/VFWallet.php [hunk 2]
```diff
@@ -123,11 +119,92 @@ public static function newTrade($requestNo, $buyerId, $buyerIdType, $buyerMobile
...
+        return [
+            'result' => false,
+            'errMsg' => $accountService->getErrorMessage()
+        ];
+    }
+
+    private static function getAccountService()
+    {
+        $config = (new VFWallet())->getDI()->getConfig();
+        $accountService = new VFAccount(
```
Comment (chaiyangyang) [inline modules/EvaPay/src/EvaPay/Models/VFWallet.php]: return new VFAccount(
+            $config->vfAccountService->service->toArray(),
+            $config->vfAccountService->http->toArray());

File: modules/EvaPay/src/EvaPay/Models/VFWallet.php [hunk 3]
```diff
@@ -123,11 +119,92 @@ public static function newTrade($requestNo, $buyerId, $buyerIdType, $buyerMobile
...
+     * @param  int $userId user id
+     * @return array              return an array, including `result` and `response` when
+     * `result` is true, `errMsg` when `result` is false.
+     */
+    public function queryBalance($userId)
+    {
+        $accountService = self::getAccountService();
+        $result = $accountService->queryBalance($userId);
+        if ($result === true) {
+            return [
```
Comment (chaiyangyang) [inline modules/EvaPay/src/EvaPay/Models/VFWallet.php]: 抽成一个方法：名字可以再想想

private function getResponse(&$result, &$accountService)
{
        if ($result === true) {
            return [
                'result' => true,
                'response' => $accountService->getResponse()
            ];
        }
        return [
            'result' => false,
            'errMsg' => $accountService->getErrorMessage()
        ];
}

File: modules/EvaPay/src/EvaPay/Services/VFAccount.php [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+    private function parseResponse($jsonArr) {
+        if (!$jsonArr || !is_array($jsonArr) ||
+            !isset($jsonArr['success'])) {
+            return false;
+        }
+
+        $this->success = $jsonArr['success'];
+        $this->response = $jsonArr;
+
+        $errCode = isset($jsonArr['errCode']) ? $jsonArr['errCode'] : '';
```
Comment (chaiyangyang) [inline modules/EvaPay/src/EvaPay/Services/VFAccount.php]: 用些常量裹住

File: modules/EvaPay/src/EvaPay/Services/VFAccount.php [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+    {
+        return $this->errMsg;
+    }
+
+    public function getErrorCode()
+    {
+        return $this->errCode;
+    }
+
+    private function setErrorMessage($errMsg)
```
Comment (chaiyangyang) [inline modules/EvaPay/src/EvaPay/Services/VFAccount.php]: errMsg不为空的时候可以拼在后面

Comment (songrgg) [inline modules/EvaPay/src/EvaPay/Services/VFAccount.php]: 维金本身的errMsg长且存在奇怪的符号，尽量不使用它，而是使用errorcode解析得到的简短的信息。

**Pull request merged**